### DISCUSSION TITLE
2FA / MFA / Multi Factor Authentication

### DISCUSSION THREAD

**Discussion by alexryanterry** [supabase/supabase#5961] [Questions] [answered]
When is 2 factor authentication going to be available? Are there any workarounds that we can use now while we wait for 2FA to come natively to Supabase?

**Comment by alexryanterry** [comment]
Sounds like there is no timeline, but there is an open issue here which is tracking this work: https://github.com/supabase/gotrue/issues/14

**Comment by J0** [comment, marked as answer]
Hey everyone, we are working on this feature in gotrue and you guys can check out the gotrue/mfa_v1 branch for the implementation details. We have also opened up an [early-sign up list](https://airtable.com/shrjcc9zi0RduHLIx) if you are using Supabase and looking to try out MFA in a project.

**Reply by alexryanterry** [reply]
Thanks for the update!
